File: Javascript/Algorithms/383.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
let canConstruct = (ransomNote, magazine) => {
    const letters = Array(26).fill(0);
    
    // Fill up with counts.
    for (let ch of magazine)
        letters[ch.codePointAt() - 97]++;
    
    // Go through and reduce counts.
    // If at any point count == 0 (and we have encountered value)
    // in ransomNote, bail.
    for(let ch of ransomNote){
        let idx = ch.codePointAt() - 97;
        if (letters[idx]-- == 0)
            return false;
    }
    return true; 
};

File: Rust/Algorithms/383.rs
impl Solution {
    pub fn can_construct(ransom_note: String, magazine: String) -> bool {
        let mut letters = vec![0; 26];

        for i in magazine.as_bytes() {
            letters[(i - 97) as usize] += 1;
        }
        for i in ransom_note.as_bytes() {
            let idx = (i - 97) as usize;
            if letters[idx] == 0 {
                return false;
            }
            letters[idx] -= 1;
        }
        true
    }
}
